### PR TITLE
Adds iterative and interval balancing comparisons

### PR DESCRIPTION
**Pull request by RyanStonebraker**
This PR introduces two algorithms for comparison:
- Iterative - **O(n<sup>2</sup>)**
  - Uses two nested loops to search for conflicts to split
- Interval Balancing - **O(n log n)**
  -  Transforms the set of intervals to a sorted list of vertices and then uses open/close balancing principles to efficiently deconflict.
interval_balancing

In addition to the new comparative algorithms, the PR updates the speed profiling jupyter notebook to include comparisons between all three algorithms.

### CODE REVIEW

**Comment by vc1492a** [comment]
@RyanStonebraker any updates on your PR?

File: comparisons/iterative.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
+def createInterval(start, finish, keys, key):
```
Comment (vc1492a) [inline comparisons/iterative.py:1]: Please change case to match: `create_interval`

File: comparisons/iterative.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    new_interval = {}
+    new_interval["start"] = start
+    new_interval["finish"] = finish
+    new_interval[key] = keys.copy()
+    return new_interval
+
+
+def getMainPermutations(intervals, key):
```
Comment (vc1492a) [inline comparisons/iterative.py:9]: Please change case to match: `get_main_permutations`

File: comparisons/iterative.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                if new_interval not in new_intervals:
+                    new_intervals.append(new_interval)
+
+            new_interval = createInterval(first_finish, last_start, start_interval[key].union(compare_interval[key]), key)
+            if new_interval not in new_intervals:
+                new_intervals.append(new_interval)
+    return conflicts, new_intervals
+
+
+def resolveConflicts(intervals, key):
```
Comment (vc1492a) [inline comparisons/iterative.py:42]: Please change case to match: `resolve_conflicts`

File: tests/test_merge_iterative_optimized.py [hunk 4]
```diff
@@ -0,0 +1,732 @@
+# Authors: Valentino Constantinou <vconstan@jpl.caltech.edu>, Asitang Mishra <asitang.mishra@jpl.caltech.edu>
+# License: Apache 2.0
+
+from comparisons.iterative_optimized import Merge
```
Comment (vc1492a) [inline tests/test_merge_iterative_optimized.py]: An import error exists here and results in tests failing, see: https://travis-ci.org/vc1492a/mieda/builds/646609867

**Review by vc1492a — CHANGES_REQUESTED** [review]

**Comment by vc1492a** [comment]
Thanks @RyanStonebraker for updating your PR, the changes look like a great start! If you could please address the above changes before merge, that would be great. 

I checked out the updated speed profiling notebook as well, and appreciate the additional plots and comparisons.

**Comment by RyanStonebraker** [comment]
https://github.com/vc1492a/mieda/pull/5/commits/35513542aa374507550e5a71a5d66bd9b7cea377 (commit Renames functions to conform to snake case) <- should address the function renaming to match snake case. Also I just pushed an update to the test cases to import the renamed interval_balancing approach!

**Review by vc1492a — APPROVED** [review]

**Pull request merged**